### PR TITLE
Add get.py for iosxe  eigrp folder containing 12 APIs for EIGRPv4 and EIGRPv6

### PR DESCRIPTION
**Pull request by cherifimehdi**
Hello, 

I noticed there is no get.py file and get apis for ios/iosxe eigrp protocol

I added get.py file at pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp  containing 12 APIs for EIGRPv4 and EIGRPv6:

               * get_eigrp_interfaces
		* get_eigrp_interfaces_v6
		* get_eigrp_interfaces_as
		* get_eigrp_interfaces_as_v6
		* get_eigrp_interfaces_peers
		* get_eigrp_interfaces_peers_v6
		* get_eigrp_interfaces_timers
		* get_eigrp_interfaces_timers_v6
		* get_eigrp_neighbors
		* get_eigrp_neighbors_v6
		* get_eigrp_router_id
		* get_eigrp_router_id_v6
		
I added all the tests at pkgs/sdk-pkg/src/genie/libs/sdk/apis/tests/iosxe/eigrp/get containing the test results for the 12 APIs in get.py

I added the changelog_add_eigrp_get.py_apis_202329121901.srt file at pkgs/sdk-pkg/changelog folder 

Thank you

### CODE REVIEW

**Comment by cherifimehdi** [comment]
Hello,

I noticed there is no get.py file and get apis for ios/iosxe eigrp protocol. So I submitted a pull request 

Thank you

**Comment by cherifimehdi** [comment]
Please @omehrabi @sowmyadn010501 could you review my contribution to genie api?

**Comment by cherifimehdi** [comment]
Hello @SohanTirpude @sumitsehgal123  @omehrabi @sowmyadn010501 could you review my contribution to genie api? Thank you

**Comment by cherifimehdi** [comment]
Hello @omehrabi  @Taarini  @SohanTirpude I updated my PR to include only 06 EIGRP get APIs supporting IPv4 and IPv6 as requested by @omehrabi  I merged all get APIs for IPv4 and IPv6.
I updated tests and changelog
Thank you for reviewing my contribution

**Review by Taarini — APPROVED** [review]

**Comment by Harishv01** [comment]
Hello @omehrabi , Please review this PR

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
+# =================================================
+# Get APIs for 'EIGRPv4 and EIGRPv6' command
+#Author: Mehdi Cherifi
+#LinkedIn: https://www.linkedin.com/in/mehdi-cherifi
+#Twitter: https://twitter.com/LocketKeepsake
+#Github: https://github.com/cherifimehdi
+# =================================================
```
Comment (omehrabi) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py:7]: please remove this

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+            interface_as=response.q.contains(as_n).get_values("interface")
+            inter_as_dict[as_n]=interface_as
+
+      return inter_as_dict
+   else:
+      log.error(f"Please, provide a valide format for vrf and/or ip!!")
+
+def get_eigrp_interfaces_timers(device, vrf='default', AS_N=None, ip='ipv4'):
+
+   """Gets hello interval and hold time for the interfaces on the device for a given vrf and AS active instance for ipv4 or ipv6 from -show ip eigrp interfaces detail- for ipv4 or -show ip eigrp interfaces detail- for ipv6
```
Comment (omehrabi) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py:82]: keep line at 80 characters

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+
+
+def get_eigrp_interfaces_as(device, vrf='default', ip='ipv4'):
+   """Gets all the interfaces on the device for a given vrf for ipv4 or ipv6 for all AS active instance from -show ip eigrp interfaces- for ipv4 by default or -show ipv6 eigrp interfaces- for ipv6
+
+      Args:
+        device (obj): Device object
+        vrf = "default" (str) : Name of the vrf
+        ip = "ipv4" (str): Protocol ip set by default to "ipv4" to change to "ipv6" 
+
```
Comment (omehrabi) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py:54]: remove extra space

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+log = logging.getLogger(__name__)
+
+
+def get_eigrp_interfaces(device, vrf='default', AS_N=None, ip='ipv4'):
+   """Gets all the interfaces on the device for a given AS and vrf for ipv4 or ipv6 from -show ip eigrp interfaces- for ipv4 by default or -show ipv6 eigrp interfaces- for ipv6
+
+      Args:
+        device (obj): Device object
+        vrf = "default" (str): Name of the vrf by default set to "default"
+        AS_N = None (int): Autonomous System
```
Comment (omehrabi) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py:21]: please provide a better name for variable

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+         else:
+            interfaces = response.q.contains(vrf).contains_key_value("eigrp_instance", str(AS_N)).contains(ip).get_values('interface')
+
+      return interfaces
+   else:
+      log.error(f"Please, provide a valide format for AS and/or vrf and ip!!")
+
+
+def get_eigrp_interfaces_as(device, vrf='default', ip='ipv4'):
+   """Gets all the interfaces on the device for a given vrf for ipv4 or ipv6 for all AS active instance from -show ip eigrp interfaces- for ipv4 by default or -show ipv6 eigrp interfaces- for ipv6
```
Comment (omehrabi) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py:48]: same here

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+
+      Returns:
+        list: List of the interfaces in -interfaces- list
+   """
+
+   if ((isinstance(AS_N, int) and isinstance (ip, str) and isinstance(vrf, str) and AS_N!=0 and ip in ['ipv4','ipv6'])):
+
+      if AS_N:
+         try:
+            response=device.parse(f"show {'ipv6' if ip!='ipv4' else 'ip'} eigrp interfaces")
```
Comment (omehrabi) [inline pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py:32]: is this right?

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py (hunk 4 above)
Comment (cherifimehdi) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py:21]: For the variable I think I used the same recommendation and it is significant (for example AS_N: Autonomous System Number)

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py (hunk 1 above)
Comment (cherifimehdi) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py]: I applied the change

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+# =================================================
+
+import logging
+import os
+from genie.metaparser.util.exceptions import SchemaEmptyParserError
+log = logging.getLogger(__name__)
+
+
+def get_eigrp_interfaces(device, vrf='default', AS_N=None, ip='ipv4'):
+   """Gets all the interfaces on the device for a given AS and vrf for ipv4 or ipv6 from -show ip eigrp interfaces- for ipv4 by default or -show ipv6 eigrp interfaces- for ipv6
```
[the remark this answers is not in the document]
Comment (cherifimehdi) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py]: I applied the change

File: pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py (hunk 3 above)
Comment (cherifimehdi) [reply pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/get.py]: I applied the change

**Comment by cherifimehdi** [comment]
Hi @omehrabi  I addressed the change. Just a remark about the variable  I think I used the same recommendation and it is significant (ASN: Autonomous System Number)

**Comment by cherifimehdi** [comment]
Sorry @omehrabi  I closed by mistake the PR

**Comment by cherifimehdi** [comment]
Now It is open

**Comment by cherifimehdi** [comment]
Hi @omehrabi  @Harishv01 This is my change as recommended by @omehrabi 
Thank you for reviewing my PR

File: pkgs/sdk-pkg/changelog/changelog_add_eigrp_get.py_apis_202329121901.rst [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+                            New
+--------------------------------------------------------------------------------
+* IOSXE
+    * Added get.py in eigrp containing 06 APIs supporting IPv4 and IPv6:
+                * get_eigrp_interfaces
+		* get_eigrp_interfaces_as
+		* get_eigrp_interfaces_peers
+		* get_eigrp_interfaces_timers
+		* get_eigrp_neighbors
+		* get_eigrp_router_id
```
Comment (omehrabi) [inline pkgs/sdk-pkg/changelog/changelog_add_eigrp_get.py_apis_202329121901.rst:11]: the indention is wrong

**Review by omehrabi — APPROVED** [review]

**Comment by cherifimehdi** [comment]
Hi @omehrabi  I have updated Changelog file. Thank you for reviewing my PR and accepting my contribution

**Pull request merged**